src/Core/Resolvers/CosmosMutationEngine.cs

```diff
@@ -234,6 +255,118 @@ private static async Task<ItemResponse<JObject>> HandleUpdateAsync(IDictionary<s
+            {
+                id = idObj?.ToString();
+            }
+
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new InvalidDataException("id field is mandatory");
+            }
+
+            if (queryArgs.TryGetValue(QueryBuilder.PARTITION_KEY_FIELD_NAME, out object? partitionKeyObj))
+            {
+                partitionKey = partitionKeyObj?.ToString();
+            }
+
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                throw new InvalidDataException("Partition Key field is mandatory");
+            }
+
+            object? item = queryArgs[MutationBuilder.ITEM_INPUT_ARGUMENT_NAME];
+
+            JObject? input;
+            // Variables were provided to the mutation
+            if (item is Dictionary<string, object?>)
+            {
+                input = (JObject?)ParseVariableInputItem(item);
+            }
+            else
+            {
+                // An inline argument was set
+                input = (JObject?)ParseInlineInputItem(item);
+            }
+
+            if (input is null)
+            {
+                throw new InvalidDataException("Input Item field is invalid");
+            }
+
+            // This would contain the patch operations to be applied on the document
+            List<PatchOperation> patchOperations = new();
+            GeneratePatchOperations(input, "", patchOperations);
+
+            int patchOperationsLimit = 10;
+            if (patchOperations.Count <= 10)
+            {
+                return (await container.PatchItemAsync<JObject>(id, new PartitionKey(partitionKey), patchOperations)).Resource;
+            }
+
+            // maximum 10 patch operations can be applied in a single patch request,
+            // Hence dividing into multiple patch request, if it is requested for more than 10 item at a time.
+            TransactionalBatch batch = container.CreateTransactionalBatch(new PartitionKey(partitionKey));
+            int numberOfBatches = -1;
+            for (int counter = 0; counter < patchOperations.Count; counter += patchOperationsLimit)
+            {
+                // Get next 10 patch operations from the list
+                List<PatchOperation> chunk = patchOperations.GetRange(counter, Math.Min(10, patchOperations.Count - counter));
+                batch = batch.PatchItem(id, chunk);
+                numberOfBatches++;
+            }
+
+            TransactionalBatchResponse response = await batch.ExecuteAsync();
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new DataApiBuilderException(
+                        message: "Failed to patch the item",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed);
+            }
+
+            return response.GetOperationResultAtIndex<JObject>(numberOfBatches).Resource;
```

Comment:
if you don't add `.Resource` here, the caller need not have conditional if/else block on operation?

```csharp
JObject result;
ItemResponse<JObject>? response = resolver.OperationType switch
                {
                    EntityActionOperation.Patch => await HandlePatchAsync(queryArgs, container),
                    EntityActionOperation.UpdateGraphQL => await HandleUpdateAsync(queryArgs, container),
                    EntityActionOperation.Create => await HandleCreateAsync(queryArgs, container),
                    EntityActionOperation.Delete => await HandleDeleteAsync(queryArgs, container),
                    _ => throw new NotSupportedException($"unsupported operation type: {resolver.OperationType}")
                };

result = response.Resource;
```

Reply:
No,` response.GetOperationResultAtIndex<JObject>(numberOfBatches)` is of type `TransactionalBatchOperationResult` which is different than `ItemResponse`.